src/main/java/com/programmers/springbootbasic/domain/voucher/Voucher.java

```diff
@@ -0,0 +1,64 @@
+package com.programmers.springbootbasic.domain.voucher;
+
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+public abstract class Voucher {
+    private final UUID voucherId;
+    private final String name;
+    private final Long condition;
+    private final LocalDateTime expirationDate;
+    private VoucherState state = VoucherState.AVAILABLE;
+
+    public Voucher(UUID voucherId, String name, LocalDateTime expirationDate) {
+        this.voucherId = voucherId;
+        this.name = name;
+        this.condition = 0L;
+        this.expirationDate = expirationDate;
+    }
+
+    public Voucher(UUID voucherId, String name, Long condition, LocalDateTime expirationDate) {
+        this.voucherId = voucherId;
+        this.name = name;
+        this.condition = condition;
+        this.expirationDate = expirationDate;
+    }
+
+    public UUID getVoucherId() {
+        return voucherId;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Long getCondition() {
+        return condition;
+    }
+
+    public LocalDateTime getExpirationDate() {
+        return expirationDate;
+    }
+
+    public VoucherState getState() {
+        if (isExpiration()) {
+            setVoucherState(VoucherState.EXPIRE);
+        }
+        return state;
+    }
+
+    public void use() {
+        setVoucherState(VoucherState.USED);
+    }
+
+    protected boolean isExpiration() {
+        LocalDateTime now = LocalDateTime.now();
+        return !now.isBefore(expirationDate);
+    }
```

Comment:
```suggestion
    protected boolean isExpiration() {
        LocalDateTime now = LocalDateTime.now();
        return now.isAfter(expirationDate);
    }
```
- 부정문 보다는 긍정문이 이해하기 쉽습니다.
- 바우처가 만료되었는지 테스트한다고 생각하면 `now` 같은 통제할 수 없는 코드가 있다면 쉽지 않을 것 같아요. 테스트 코드에서 expirationDate를 2023년 01월 01일로 설정한다고 생각해볼게요. 그러면 now가 2022년 12월 10일인 경우와 now가 2023년 06월 01인 경우 반환값이 달라질 거에요.
 
https://jojoldu.tistory.com/676